### PR TITLE
[MM-33462] Extend manifest type

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
`AppManifest` was missing a couple of fields are are both useful for the webapp and for app developers.

I intentionally leave the fields out that are purely for the backend like callbacks and the AWS information.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-33462

### CODE REVIEW

**Review by mickmister — APPROVED** [review]
Found one typo, LGTM otherwise :+1:

**Comment by hanzei** [comment]
@jfrerich Gentle reminder to take a look when you have the time.

**Comment by amyblais** [comment]
@jfrerich @catalintomai Kind reminder for PR review.

File: packages/mattermost-redux/src/types/apps.ts [hunk 1]
```diff
@@ -4,11 +4,38 @@
...
 // Apps Framework feature is experimental, and the contents of this file are
 // susceptible to breaking changes without pushing the major version of this package.
+
+export enum AppType {
+    HTTP = 'http',
+    AWSLambda = 'aws_lambda',
+    Builtin = 'builtin',
+}
+
+export enum Permission {
```
Comment (jfrerich) [inline packages/mattermost-redux/src/types/apps.ts:14]: Suggest that we use the plural to match the `Locations` enum
```suggestion
export enum Permissions {
```

**Review by jfrerich — APPROVED** [review]
One small suggestion.

File: packages/mattermost-redux/src/types/apps.ts (hunk 1 above)
Comment (hanzei) [reply packages/mattermost-redux/src/types/apps.ts:14]: https://www.typescriptlang.org/docs/handbook/enums.html is using singular names for enums. https://docs.microsoft.com/en-us/previous-versions/dotnet/netframework-1.1/4x252001(v=vs.71)?redirectedfrom=MSDN also states

> Use a singular name for most Enum types, but use a plural name for Enum types that are bit fields.


Hence, I'm leaving it as singular.

**Pull request merged**